aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -760,6 +763,8 @@ public class AwsProperties implements Serializable {
   private String restSecretAccessKey;
   private String restSessionToken;
 
+  private String sqsQueueUrl;
```

Comment:
Start from this, let's separate the properties from `AwsProperties` and put it in a different class like `SqsMetricsReporterProperties`. We will gradually also separate other properties to complete the refactoring so that we no longer depend on the client factory for all AWS clients

Reply:
Started to separate out the properties. I have kept a base class for common properties across all clients.